great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -776,7 +776,20 @@ def _update_fluent_datasource(
 
         self.datasources[datasource_name] = update_datasource
 
-    def _delete_fluent_datasource(self, datasource_name: str) -> None:
+    def _delete_fluent_datasource(
+        self, datasource_name: str, _call_store: bool = True
+    ) -> None:
+        """
+        _call_store = False allows for local deletes without deleting the persisted storage datasource.
+        This should generally be avoided.
+        """
+        datasource = self.datasources.get(datasource_name)
+        if datasource:
+            if self._datasource_store.cloud_mode and _call_store:
+                self._datasource_store.delete(datasource)  # type: ignore[arg-type] # Could be a LegacyDatasource
```

Comment:
We have work scoped to use a `Store` for all FDS persistence logic and avoid this branching.